tests/FunctionalTestCase.php

```diff
@@ -334,9 +334,6 @@ protected function skipIfChangeStreamIsNotSupported()
                 if (! $this->isShardedClusterUsingReplicasets()) {
                     $this->markTestSkipped('$changeStream is only supported with replicasets');
                 }
-
-                // Temporarily skip tests because of an issue with change streams in the driver
-                $this->markTestSkipped('$changeStreams currently don\'t on replica sets');
```

Comment:
The PR itself LGTM, but I realized that removing this skip re-introduced the test failures.

I assume those were not happening in previous Travis builds since this skip was first introduced in https://github.com/mongodb/mongo-php-library/commit/2008bc7733c993d80055accc62cddd6d50b5d0ac as part of https://github.com/mongodb/mongo-php-library/pull/655?

We can discuss offline about whether to merge as-is for now and defer sorting out the double-free for replica set and sharded replica set tests in [PHPLIB-432](https://jira.mongodb.org/browse/PHPLIB-432).

Reply:
Yes and no. Not sure what made the errors appear consistently, but the skip only affected sharded clusters, not the replicaset build. With this branch, the replicaset build always fails while the build on a sharded cluster at least passes sometimes.

Since this entire PR deals with fixing tests on sharded clusters, I don't see a reason why to merge this without running the tests on a sharded cluster. I'd first like to investigate the cause for the double free some more before just plain skipping those tests that cause the failures.